src/composed/message/types.rs

```diff
@@ -397,12 +397,9 @@ impl Message {
     ) -> Result<Self> {
         let data = self.to_bytes()?;
 
-        let edata = Edata::SymEncryptedProtectedData(SymEncryptedProtectedData::encrypt_with_rng(
-            rng,
-            alg,
-            &session_key,
-            &data,
-        )?);
+        let edata = Edata::SymEncryptedProtectedData(
```

Comment:
- Add a check that the `esk` parameter contains `Esk` that are all consistently esk v3/4 or esk v6, and then
- use the appropriate seipd version

Reply:
Note to self: renamed the existing functions to `*_seipdv1` to make it explicit that they produce SEIPDv1 encrypted data packets.

Adding analogous mid-level APIs that produce SEIPDv2 encrypted data packets is a separate task.